src/cdk/v2/destinations/zoho/transformRecordV2.ts

```diff
@@ -337,6 +400,8 @@ const processRecordInputsV2 = async (inputs: ZohoRouterIORequest[], destination?
 
   const response: unknown[] = [];
   const errorResponseList: unknown[] = [];
+  // Phase 1: Process all events, queue deletions (don't execute COQL yet)
+  const deletionQueue: DeletionQueueItem[] = [];
```

Comment:
@maheshkutty, let's simplify the implementation to first group the input events based on the "action".